crates/ark/src/lsp/help.rs

```diff
@@ -57,7 +81,22 @@ impl RHtmlHelp {
         // parse as html
         let contents = contents.to::<String>()?;
         let html = Html::parse_document(contents.as_str());
-        Ok(Some(Self { html }))
+
+        Ok(Some(html))
+    }
+
+    /// Is this a help page for a function?
+    ///
+    /// Uses a heuristic of looking for a `Usage` section to determine if this looks like
+    /// function help or not. Can't look for `Arguments`, as some functions don't have
+    /// any!
+    fn is_function(x: &Html) -> bool {
+        // Find all h3 headers in the document
+        let selector = Selector::parse("h3").unwrap();
+        let mut headers = x.select(&selector);
+
+        // Do any have a usage section?
+        headers.any(|header| header.html() == "<h3>Usage</h3>")
```

Comment:
The `Usage` trick

Reply:
I guess the html structure is stable enough? Can we analyse the Rd instead, perhaps eventually? It feels like we should be able to do better (though it's fine to do it this way now as a workaround, I'm just discovering this problem space and thinking aloud).

Reply:
> Can we analyse the Rd instead

Possibly yea, we'd have to return the `Rd` path (or contents) from `.ps.help.getHtmlHelpContents()` as well as the HTML contents (and maybe change the function name), and then store the contents in `RHtmlHelp` alongside the HTML. It seems like both installed help and dev help have access to a real `Rd` file on disk, so that's good.

And then add our own Rd parser on top of that to analyze it